### PR TITLE
feat: add sms provider seven

### PR DESCRIPTION
**Pull request by matthiez**
Hi,
I added a new a new SMS provider [seven](https://www.seven.io).

Hope this helps!

### CODE REVIEW

File: src/providers/sms/seven.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  async send (request: SmsRequestType): Promise<string> {
+    const { from, text, to, type, ttl, messageClass } = request.customize ? (await request.customize(this.id, request)) : request
+    const params = {
+      flash: messageClass === 0,
+      from,
+      text,
+      to,
+      ttl,
+      unicode: type === 'unicode'
+    }
```
Comment (BDav24) [inline src/providers/sms/seven.js:26]: It seems they use `0` or `1` in place of booleans in [their documentation](https://www.seven.io/en/docs/gateway/http-api/sms-dispatch/#optional-parameters):
```suggestion
    const params = {
      flash: messageClass === 0,
      from,
      text,
      to,
      ttl,
      unicode: type === 'unicode'
    }
```
```suggestion
    const params = {
      flash: messageClass === 0 ? 1 : 0,
      from,
      text,
      to,
      ttl,
      unicode: type === 'unicode' ? 1 : 0
    }
```

File: lib/providers/sms/index.js
Comment (BDav24) [inline lib/providers/sms/index.js:1]: You can remove this file from the PR, it'll be generated during deployment.

File: __tests__/providers/sms/seven.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+      'Content-Length': ['102'],
+      'Content-Type': ['application/json'],
+      SentWith: ['Notifme'],
+      'User-Agent': ['notifme-sdk/v1 (+https://github.com/notifme/notifme-sdk)'],
+      'X-Api-Key': ['apiKey']
+    })
+  }))
+  expect(mockHttp.body).toEqual(
+    '{"flash":false,"from":"Notifme","text":"Hello John! How are you?","to":"+15000000001","unicode":false}'
+  )
```
Comment (BDav24) [inline __tests__/providers/sms/seven.js:74]:
```suggestion
    headers: expect.objectContaining({
      Accept: ['application/json'],
      'Content-Length': ['94'],
      'Content-Type': ['application/json'],
      SentWith: ['Notifme'],
      'User-Agent': ['notifme-sdk/v1 (+https://github.com/notifme/notifme-sdk)'],
      'X-Api-Key': ['apiKey']
    })
  }))
  expect(mockHttp.body).toEqual(
    '{"flash":0,"from":"Notifme","text":"Hello John! How are you?","to":"+15000000001","unicode":0}'
  )
```

File: __tests__/providers/sms/seven.js [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+  mockResponse(200, JSON.stringify(apiResponse))
+  await sdk.send({
+    sms: {
+      ...request.sms,
+      customize: async (provider, request) => ({ ...request, text: 'Hello John! How are you?' })
+    }
+  })
+  expect(mockHttp).lastCalledWith(expect.objectContaining({
+    headers: expect.objectContaining({ 'Content-Length': ['103'] })
+  }))
```
Comment (BDav24) [inline __tests__/providers/sms/seven.js:93]:
```suggestion
  await sdk.send({
    sms: {
      ...request.sms,
      customize: async (provider, request) => ({
        ...request,
        text: 'a totally new message',
        messageClass: 0,
        type: 'unicode'
      })
    }
  })
  expect(mockHttp.body).toEqual(
    '{"flash":1,"from":"Notifme","text":"a totally new message","to":"+15000000001","unicode":1}'
  )
```

**Review by BDav24 — COMMENTED** [review]
Hi @matthiez, thanks for your PR!
It looks good to me, just a comment about boolean parameters.

**Comment by BDav24** [comment]
Merged in https://github.com/notifme/notifme-sdk/pull/151

File: src/providers/sms/seven.js (hunk 1 above)
Comment (matthiez) [reply src/providers/sms/seven.js:26]: Both works :)

**Pull request closed without merging**